mlir/tools/rocmlir-lib/CMakeLists.txt

```diff
@@ -45,57 +47,344 @@ if(BUILD_FAT_LIBROCKCOMPILER)
   set(PACKAGE_NAME ${CMAKE_PROJECT_NAME_LOWER})
 
   set(LIBRARY_NAME rockCompiler)
-  function(combine_archives output_archive)
-    set(mri_file ${CMAKE_CURRENT_BINARY_DIR}/${output_archive}.mri)
-    set(full_output_path ${ROCMLIR_LIB_DIR}/lib${output_archive}.a)
-    set(output_archive_dummy_file ./${output_archive}.dummy.cpp)
+  set(COMPONENT_NAME ${LIBRARY_NAME})
+
+  set(__rocmlir_targets
```

Comment:
I'm still extremely dubious about this - what, exactly, are we gaining from hard-coding the list of transitive dependencies like this? 

If we're building static libraries that aren't needed, could you point out which ones, so we can get them out of the dependency tree?

Otherwise ... what's this solving?

Reply:
I changed the hardcoded list of files to `file(GLOB ...)`. However, please consider the warning given by the authors of CMake, about globbing.

> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

Source: https://cmake.org/cmake/help/v3.26/command/file.html#glob
The same applies to globbing using a shell from CMake. That is even worse because CMake does not know we are doing such a thing.